tests/solidity/test_open_short.py

```diff
@@ -0,0 +1,199 @@
+"""Test opening a short in hyperdrive"""
+import unittest
+import decimal
+
+import elfpy.agents.agent as agent
+import elfpy.markets.hyperdrive.hyperdrive_market as hyperdrive_market
+import elfpy.pricing_models.hyperdrive as hyperdrive_pm
+import elfpy.types as types
+import elfpy.time as time
+
+
+class TestOpenShort(unittest.TestCase):
+    """
+    Test opening a short in hyperdrive, with the following cases:
+        open a short of 0 size (failure)
+        open a short of extreme size (failure)
+    """
+
+    contribution: float = 500_000_000
+    target_apr: float = 0.05
+    term_length: int = 365
+    alice: agent.Agent
+    bob: agent.Agent
+    celine: agent.Agent
+    hyperdrive: hyperdrive_market.Market
+    block_time: time.BlockTime = time.BlockTime()
+
+    def setUp(self):
+        self.alice = agent.Agent(wallet_address=0, budget=self.contribution)
+        self.bob = agent.Agent(wallet_address=1, budget=self.contribution)
+        self.celine = agent.Agent(wallet_address=2, budget=self.contribution)
+        pricing_model = hyperdrive_pm.HyperdrivePricingModel()
+        market_state = hyperdrive_market.MarketState()
+        self.hyperdrive = hyperdrive_market.Market(
+            pricing_model=pricing_model,
+            market_state=market_state,
+            block_time=self.block_time,
+            position_duration=time.StretchedTime(
+                days=self.term_length,
+                time_stretch=pricing_model.calc_time_stretch(self.target_apr),
+                normalizing_constant=self.term_length,
+            ),
+        )
+        _, agent_deltas = self.hyperdrive.initialize(self.alice.wallet.address, self.contribution, 0.05)
+        self.alice.wallet.update(agent_deltas)
+
+    def verify_open_short(
+        self,
+        user: agent.Agent,
+        market_state_before: hyperdrive_market.MarketState,
+        base_amount: float,  # max loss in base transferred from user to hyperdrive
+        unsigned_bond_amount: float,  # number of PTs shorted
+        maturity_time: int,  # maturity of the opened short
+        apr_before: float,
+    ):  # pylint: disable=too-many-arguments
+        """
+        Verify that the market state is updated correctly after opening a short.
+        Contains the following checks:
+        - Hyperdrive received the max loss and that Bob received the short tokens
+        - initializing the pool to the target APR worked
+        - opening a short doesn't make the APR go down
+        - reserves are updated correctly for: shares, bonds, LP tokens, share price
+            longs_outstanding, long_average_maturity_time, long_base_volume, long_base_volume_checkpoints,
+            shorts_outstanding, short_average_maturity_time, short_base_volume, short_base_volume_checkpoints
+        """
+        # Total amount of base tokens locked in Hyperdrive
```

Comment:
Why is this one commented? Can you add a TODO noting when it will be fixed?

Reply:
added TODO: this can be enabled if we add a metric that measures total TVL deposited into the smart contract